src/Adapter/MSTest.CoreAdapter/ObjectModel/UnitTestElement.cs

```diff
@@ -114,10 +114,21 @@ internal TestCase ToTestCase()
                 this.TestMethod.FullClassName,
                 this.TestMethod.Name);
 
-            TestCase testCase = new TestCase(fullName, TestAdapter.Constants.ExecutorUri, this.TestMethod.AssemblyName);
-            testCase.DisplayName = string.IsNullOrEmpty(this.DisplayName) ? this.TestMethod.Name : this.DisplayName;
+            TestCase testCase;
 
+            if (string.IsNullOrWhiteSpace(this.TestMethod.ManagedType) || string.IsNullOrWhiteSpace(this.TestMethod.ManagedMethod))
```

Comment:
Similar comment in vstest repo. Consider creating a shared helper that is used everywhere to determine if we have `ManagedType`/`ManagedMethod`